coalib/io/FileFactory.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('name', 'timestamp')
+class FileFactory:
+    """
+    The ``FileFactory`` is a wrapper class whose objects contain the
+    following information about the file:
+
+        * The filename (absolute file path).
+        * The modification timestamp of the file.
+        * The file in raw format.
+        * The file in the form of a string UTF-8 decoded from the raw format.
+        * The lines of the file in the form of a tuple.
+
+    The equality of the object is checked just based on its filename and
+    timestamp.
+
+    To initialize a ``FileFactory`` object for a file:
+
+    >>> import tempfile
+    >>> temp = tempfile.NamedTemporaryFile()
+    >>> temp.write(bytes('This is a test file.', 'UTF-8'))
+    20
+    >>> temp.seek(0)
+    0
+    >>> fp = FileFactory(temp.name)
+
+    File indices start with zero in case of
+    ``FileFactory`` objects. To retrieve a single line:
+
+    >>> fp.get_line(0)
+    'This is a test file.'
+
+    Get all the lines in the file as a list:
+
+    >>> fp.lines
+    ('This is a test file.',)
+
+    Get the file contents in raw mode:
+
+    >>> fp.raw
+    b'This is a test file.'
+
+    Get the file as a string:
+
+    >>> fp.string
+    'This is a test file.'
+
+    Get the filename:
+
+    >>> fp.name == temp.name
+    True
+
+    Make the object into an iterable when required
+
+    >>> for line in fp:
+    ...     print(line)
+    This is a test file.
+    """
```

Comment:
Delete the temp file again ;)

Reply:
If you delete it, leave another whitespace before last code, so the user sees that's just cleanup and not belonging to the iterator functionality:
```
>>> ...
... print(line)
This is a test file.

>>> temp.delete()
```